### DISCUSSION TITLE
How to make pull request between several branches?

### DISCUSSION THREAD

**Discussion by plamen-ivanov1** [community/community#64635] [Pull Requests]
### Select Topic Area

Question

### Body

Hello,

I'm facing a situation where I need to create pull requests that involve multiple branches in my GitHub repository. I would appreciate some guidance on how to handle this scenario effectively.

#### Here's the context:

- I have a main branch, let's call it "main," which represents the stable version of my project.
- I also have several feature branches, each dedicated to a specific feature or bug fix.
- Now, I want to create pull requests that merge changes from these feature branches into the main branch. However, since the changes are spread across different branches, I'm a bit unsure about the best approach.

### Here are a few questions I have:

- What's the recommended way to create pull requests involving multiple branches?
- Should I create separate pull requests for each feature branch, or is there a way to bundle them together?
- How do I ensure that the changes from different pull requests don't conflict with each other?
- Are there any best practices or tools that can help streamline this process?

I'm also wondering if there are any potential pitfalls or challenges I should be aware of when dealing with pull requests that span multiple branches.

Any advice, tips, or step-by-step instructions you can provide would be greatly appreciated. Thank you for your time and assistance!

**Comment by webguru11124** [comment]
I think One recommended approach is to create separate pull requests for each feature branch, as this allows for easier review and testing of each individual change. However, you can also bundle them together if they are related and make sense to be reviewed as a group.

To ensure that changes from different pull requests don't conflict with each other, it's important to regularly merge the main branch into each feature branch to keep them up-to-date. Additionally, using a tool like GitHub's merge conflict resolution can help identify and resolve any conflicts that arise.

Some best practices for managing pull requests involving multiple branches include regularly communicating with your team members about changes and updates, using descriptive commit messages, and testing changes thoroughly before merging them into the main branch.

There are also several tools available that can help streamline the process, such as GitHub's pull request templates, automated testing tools, and code review tools like CodeClimate or Codacy.

As for potential pitfalls or challenges, it's important to be aware of the potential for merge conflicts and to have a plan in place for resolving them. Additionally, it's important to ensure that each feature branch is thoroughly tested before merging into the main branch to avoid introducing bugs or other issues.

**Comment by queenofcorgis** [comment]
GitHub's [Acceptable Use Policies](https://docs.github.com/en/site-policy/acceptable-use-policies/github-acceptable-use-policies) prohibits coordinated or inauthentic activity like rapid questions and answers. As a result, we'll be unmarking the answer and locking this post. Please note any future violations may result in a temporary or indefinite block from the Community. Thanks for understanding.
